apps/server/src/utils/arrayUtils.ts

```diff
@@ -61,3 +61,15 @@ export const sortArrayByProperty = <T>(arr: T[], property: string): T[] => {
     return a[property] - b[property];
   });
 };
+
+/**
+ * @description Creates a nested object with keys from an array and assigns the `value` to the last key
+ * @param {array} arr - array to be nested
+ * @param {string} value - value to assign
+ * @returns {object | null} nested object or null if no object was created
+ */
+export const nestedObjectFromArray = (arr: string[], value?: unknown): object | null => {
+  const obj = arr.reduceRight((result, key) => ({ [key]: result }), value);
+  if (typeof obj === 'object') return obj;
+  return null;
```

Comment:
I dont think we need this generic function. our function should take a path and produce an object. This is only  useful in the context of the APIs and we shouldnt motivate to reuse anywhere else

That would also make our tests more valuable since they would include real paths